homework/index.js

```diff
@@ -29,19 +29,127 @@
     });
     return elem;
   }
-
+  function singlePageApplication(arr) {
+    //header
+    const div = createAndAppend('div', root, {
+      id: 'header-div',
+    });
+    const h1 = createAndAppend('h1', div, {
+      text: 'HYF Repositories',
+      id: 'header',
+    });
+    const selectMenu = createAndAppend('select', div, {
+      id: 'select-menu',
+    });
+    for (let i = 0; i < arr.length; i++) {
+      const option = createAndAppend('option', selectMenu, {
+        text: arr[i].name,
+        value: i,
+      });
+    }
+    //right and left part
+    const containerDiv = createAndAppend('div', root, {
+      id: 'container',
+      class: 'container',
+    });
+    //right part
+    const rightDiv = createAndAppend('div', containerDiv, {
+      id: 'right-div',
+      class: 'contained',
+    });
+    const tHeads = ['Repository: ', 'Description: ', 'Forks: ', 'Update: '];
+    const table = createAndAppend('table', rightDiv, {
+      id: 'tableOfInformation',
+    });
+    for (let i = 0; i < tHeads.length; i++) {
+      var firstTime = ['name', 'description', 'forks', 'updated_at'];
+      const tr = createAndAppend('tr', table, {});
+      const tableHead = createAndAppend('th', tr, {
+        text: tHeads[i],
+      });
+      if (i === 0) {
+        const tableData = createAndAppend('td', tr, {
+          id: `tableData.${i}`,
+        });
+        const link = createAndAppend('a', tableData, {
+          href: arr[i].html_url,
+          text: arr[i].name,
+          target: '_blank',
+        });
+      } else {
+        const tableData = createAndAppend('td', tr, {
+          text: arr[0][firstTime[i]],
+          id: `tableData.${i}`,
+        });
+      }
+    }
+    selectMenu.addEventListener('change', event => {
+      for (let i = 0; i < 4; i++) {
+        const changed = document.getElementById(`tableData.${i}`);
+        if (i === 0) {
+          changed.innerHTML = `<a target = "_blank" href = ${arr[event.target.value].html_url}> ${
```

Comment:
Syntax with all the spaces looks weird and might break things, it should be like this; `target="_blank" href="${}"`